### PR TITLE
DDF-4445 Add Plugin support for location serialization

### PR DESCRIPTION
**Pull request by middlets719**
#### What does this PR do?
Adding Plugin support to the serializing and deserializing of the location model to allow new search types to be added and still preserve user query selects on reloads
#### Who is reviewing it? 
@Schachte @zta6 @gordocanchola @adimka 
#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@djblue

#### How should this be tested?
Ingest Gazetteer information with `gazetteer:update "$ddf.data/countries.geo.json"`
Regression test Query Preservation:
Create and save separate location search with each of the following types, Lines, Polygon, Point, BBox, and Keyword.  
Reopen each search and verify the location information is the same as when it was saved.
#### What are the relevant tickets?
For Jira:
[](https://codice.atlassian.net/browse/)

For GH Issues:
Fixes: #4445 ____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/plugins/location-serialization.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = v => v
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/plugins/location-serialization.js]:
```
exports.Serializers = (v) => v
exports.Deserializers = (v) => v
```

Comment (Schachte) [reply ui/packages/catalog-ui-search/src/main/webapp/plugins/location-serialization.js]: This is better than 2 separate plugins actually

**Comment by Schachte** [comment]
✏️ Should add a comment to note that this is temporary since we have a plan to remove location-old, so no one should be heavily relying on this implementation.

**Comment by middlets719** [comment]
build now

File: ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.view.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ const filterToLocationOldModel = filter => {
     typeof filter.value === 'object' ? filter.value.value : filter.value
 
   // for backwards compatability with wkt
-  if (typeof filterValue === 'string') {
+  if (filterValue && typeof filterValue === 'string') {
```
Comment (Schachte) [inline ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.view.js]: This is redundant

**Comment by Schachte** [comment]
build now

**Comment by vinamartin** [comment]
@middlets719 or whichever committer merges this, please make sure to update the commit title and message to reflect the ticket #. Thank you!

**Comment by middlets719** [comment]
build now

**Comment by Schachte** [comment]
screen shot 2019-02-28 at 5 10 16 pm

Hero looks good when persisting different types and properly serializing back on refresh within the workspace. Tried advanced and basic query construction and queries are working successfully.

**Pull request merged**